menas/ui/components/login/loginMain.controller.js

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+sap.ui.define([
+  "sap/ui/core/mvc/Controller",
+  "sap/m/MessageToast"
+], function (Controller, MessageToast) {
+  "use strict";
+
+  return Controller.extend("components.login.loginMain", {
+    loginForm: {},
+
+    /**
+     * Called when a controller is instantiated and its View controls (if
+     * available) are already created. Can be used to modify the View before it
+     * is displayed, to bind event handlers and do other one-time
+     * initialization.
+     *
+     * @memberOf components.login.loginMain
+     */
+    onInit: function () {
+      this._router = sap.ui.core.UIComponent.getRouterFor(this);
+      let model = {username: "", password: "", submit: "Login"};
+
+      this.loginForm = this.byId("loginForm");
+      this.loginForm.setModel(new sap.ui.model.json.JSONModel(model), "login");
+
+      this.byId("loginSubmit").attachPress(this.onLoginSubmit, this);
+    },
+
+    onLoginSubmit: function (oEvent) {
+      let oData = this.loginForm.getModel("login").oData;
+
+      this._resetLoginFormState();
+      if (this._validateLogin(oData)) {
+        this._login(oData, this.byId("loginSubmit"))
+      }
+    },
+
+    _resetFieldState: function (sField) {
+      this.byId(sField).setValueState(sap.ui.core.ValueState.None);
+      this.byId(sField).setValueStateText("");
+    },
+
+    _resetLoginFormState: function () {
+      this._resetFieldState("username");
```

Comment:
Here comes the question/discussion. Do we want to be the guys that make you re-type even the username?

Reply:
You saying extract it? I can do that.